### DISCUSSION TITLE
Compute shaders on Intel Mac

### DISCUSSION THREAD

**Discussion by SmnTin** [bevyengine/bevy#7191] [Q&A]
I can't run even simplest examples on my Intel Macbook:
```console
$ cargo run --example compute_shader_game_of_life
    Finished dev [unoptimized + debuginfo] target(s) in 0.58s
     Running `target/debug/examples/compute_shader_game_of_life`
2023-01-14T16:01:59.318459Z  INFO bevy_render::renderer: AdapterInfo { name: "Intel(R) Iris(TM) Plus Graphics 645", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
2023-01-14T16:01:59.639855Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "MacOS 13.1 ", kernel: "22.2.0", cpu: "Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz", core_count: "4", memory: "16.0 GiB" }
2023-01-14T16:01:59.789431Z ERROR wgpu::backend::direct: Handling wgpu errors as fatal by default    
thread 'Compute Task Pool (2)' panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_bind_group
    the adapter does not support read access for storages texture of format Rgba8Unorm

', /Users/smntin/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.14.2/src/backend/direct.rs:2403:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/smntin/bla/bevy/crates/bevy_tasks/src/task_pool.rs:306:45
```

I did some experimentation previously trying to rewrite the example shader in a way it could run but haven't succeeded so far.

The same applies to any tutorial on compute shaders I find on the internet, however, googling the error message itself doesn't help. 

According to [this discussion](https://github.com/gpuweb/gpuweb/issues/513) such read-write storage should be supported on Metal. 

This seems to be an issue with `wgpu` rather than with `bevy` but I feel that here I can find help more easily :) In the end, I try to use compute shaders with `bevy` and not with `wgpu` directly. 

Maybe some prior configuration is required to make the shader work on Metal which wasn't included in the example to keep it concise.

**Comment by regnarock** [comment]
Just faced the exact same issue on my side!
